src/controller.ts

```diff
@@ -0,0 +1,135 @@
+import { name, version } from '../package.json';
```

Comment:
Moved all the real logic out of index.ts and into a controller.ts for testibility and code readability